### DISCUSSION TITLE
如何只要一个编辑框，自带语法检测和格式化

### DISCUSSION THREAD

**Discussion by aogg** [DTStack/molecule#901] [Ideas]
如何只要一个编辑框，自带语法检测和格式化

**Comment by resetsix** [comment]
编辑器使用使用微软的 [monaco-editor](https://github.com/microsoft/monaco-editor) 或社区封装的[@monaco-editor/react](https://github.com/suren-atoyan/monaco-react)。

实现`代码高亮`、`语法校验`、`自动补全`功能只需要把[monaco-sql-languages](https://github.com/DTStack/monaco-sql-languages)集成在编辑器中就行。

对于格式化参考：
- [sql-formatter](https://github.com/sql-formatter-org/sql-formatter)
